### PR TITLE
Bug 1189144 - fx a mock cleanup test

### PR DESCRIPTION
**Pull request by pdehaan**
r? @silne30 @JohanLorenzo

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5f592f8a8ec275a31216e87ae5b775d03f60c86f

File: apps/system/fxa/test/marionette/fxa_screen_flow_test.js [hunk 1]
```diff
@@ -14,19 +13,24 @@ marionette('Firefox Accounts Screen Flow Test (UITest app)', function() {
...
       });
 
-  setup(function() {
-    fxaUser= new FxAUser();
+  setup(function(done) {
+    fxaUser = new FxAUser();
     app = new FxA(client);
     Server.create(FxA.SERVER_ARGS, function (err, _server) {
+      if (err) {
+        console.error(err);
```
Comment (JohanLorenzo) [inline apps/system/fxa/test/marionette/fxa_screen_flow_test.js]: Do we need to continue the test even if the server is not created? If not, we could add below:
```js
throw err;
```

File: apps/system/fxa/test/marionette/fxa_screen_flow_test.js [hunk 2]
```diff
@@ -14,19 +13,24 @@ marionette('Firefox Accounts Screen Flow Test (UITest app)', function() {
...
               config.SERVER_PORT + '/' +
               config.SERVER_PATH
-            }
+          }
         }
       });
 
-  setup(function() {
-    fxaUser= new FxAUser();
+  setup(function(done) {
```
Comment (JohanLorenzo) [inline apps/system/fxa/test/marionette/fxa_screen_flow_test.js]: Having the setup async is great :+1:

File: apps/system/fxa/test/marionette/lib/fxa.js [hunk 3]
```diff
@@ -63,8 +62,8 @@ FxA.Selectors = {
     pwSetInput: '#fxa-pw-set-input',
     pwRefresh: '#fxa-pw-input-refresh',
     COPPAElementId: '#fxa-coppa',
-    COPPASelectId: 'fxa-age-select',
-    COPPAOptionVal: '1990 or earlier',
+    COPPASelectId: '#fxa-age-select',
```
Comment (JohanLorenzo) [inline apps/system/fxa/test/marionette/lib/fxa.js]: Out of curiosity, what does COPPA stand for?

File: apps/system/fxa/test/marionette/lib/fxa.js [hunk 4]
```diff
@@ -120,6 +122,15 @@ FxA.prototype = {
...
+      var client = this.client;
+
+      client.waitFor(function () {
+        return client.findElement(FxA.Selectors.moduleNext)
+          .scriptWith(function (el) {
+            return el.getAttribute('disabled') !== 'disabled';
+          });
+      });
+
       this.client.helper
```
Comment (JohanLorenzo) [inline apps/system/fxa/test/marionette/lib/fxa.js]: Nit: `client` instead of `this.client`

**Comment by JohanLorenzo** [comment]
@pdehaan Looks great! A few nits and questions left in the PR

File: apps/system/fxa/test/marionette/lib/fxa.js (hunk 3 above)
Comment (pdehaan) [inline apps/system/fxa/test/marionette/lib/fxa.js]: "Children's Online Privacy Protection Act"; https://en.wikipedia.org/wiki/Children%27s_Online_Privacy_Protection_Act

Basically we need to do an age check to make sure we're doing the right thing.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=209f628cb9e2a509988acad1db7653259c3c4ac7

**Comment by pdehaan** [comment]
@JohanLorenzo @silne30 Any chance we can get a :+1: and a merge?
Looks like I lack the correct "write access" to self-merge.

:shipit:

**Comment by silne30** [comment]
PR Looking good! I will go ahead and +1 for merging goodness!

**Pull request merged**